### ISSUE TITLE
Custom types with procmacro

### ISSUE THREAD

**Issue by martinmose** [mozilla/uniffi-rs#1877]
Hello,

I've been exploring custom type conversions in UniFFI, specifically the example given here: [Custom Types in the Bindings Code](https://mozilla.github.io/uniffi-rs/udl/custom_types.html#custom-types-in-the-bindings-code).

Here's the code I'm working with:

```rust
#[derive(uniffi::Record)]
pub struct Handle {
    pub url: i64,
}

impl UniffiCustomTypeConverter for Handle {
    type Builtin = i64;

    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
        Ok(Handle{url: val})
    }

    fn from_custom(obj: Self) -> Self::Builtin {
        obj.url
    }
}
```

And my `uniffi.toml` looks like this:

```
[bindings.kotlin.custom_types.Url]
type_name = "URL"
imports = [ "java.net.URL" ]
into_custom = "URL({})"
from_custom = "{}.toString()"

[bindings.swift.custom_types.Url]
type_name = "URL"
imports = [ "Foundation" ]
into_custom = "URL({})"
from_custom = "{}.toString()"
```

However, in the generated `.swift` file, I still see:
```
public struct Handle {
    public var url: Int64
}
```

I expected to see:
```
public struct Handle {
    public var url: URL
}
```

We're keen to utilize custom types so that we can use `Date`, `URL`, `Decimal`, etc., in Swift.

So, in short, does the above approach only work with UDL and not with proc macros? 🤔

Thank you for your help!

**Comment by mhammond** [comment]
The code you are working with uses Handle/i64 but the .toml file is using URL/String.

**Comment by martinmose** [comment]
Firstly, I tried the following implementation:

```
#[derive(uniffi::Record)]
pub struct Handle {
    pub url: String,
}

impl UniffiCustomTypeConverter for Handle {
    type Builtin = String;

    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
        Ok(Handle{url: val})
    }

    fn from_custom(obj: Self) -> Self::Builtin {
        obj.url
    }
}
```

However, I wanted to align my implementation with the example provided in the documentation:
https://mozilla.github.io/uniffi-rs/udl/custom_types.html#custom-types-in-the-bindings-code

😬 

But same result with the String version.

**Comment by martinmose** [comment]
Sorry, I found: https://mozilla.github.io/uniffi-rs/proc_macro/index.html#the-unifficustom_type-and-unifficustom_newtype-macros

It works now. Sorry for confusion 😭
